engine/table/src/main/java/io/deephaven/engine/table/impl/locations/local/KeyValuePartitionLayout.java

```diff
@@ -3,39 +3,27 @@
 //
 package io.deephaven.engine.table.impl.locations.local;
 
-import io.deephaven.base.verify.Require;
 import io.deephaven.engine.table.Table;
-import io.deephaven.api.util.NameValidator;
-import io.deephaven.engine.table.impl.locations.TableDataException;
 import io.deephaven.engine.table.impl.locations.TableLocationKey;
 import io.deephaven.engine.table.impl.locations.impl.TableLocationKeyFinder;
 import io.deephaven.engine.table.ColumnSource;
 import org.jetbrains.annotations.NotNull;
-import org.jetbrains.annotations.Nullable;
 
-import java.io.File;
-import java.io.IOException;
-import java.nio.file.*;
-import java.nio.file.attribute.BasicFileAttributes;
 import java.util.*;
+import java.net.URI;
+import java.nio.file.Path;
 import java.util.function.BiFunction;
 import java.util.function.Consumer;
-import java.util.function.Predicate;
-import java.util.function.Supplier;
 
 /**
- * {@link TableLocationKeyFinder Location finder} that will traverse a directory hierarchy and infer partitions from
- * key-value pairs in the directory names, for example:
- * 
- * <pre>
- * tableRootDirectory/Country=France/City=Paris/parisData.parquet
- * </pre>
- * 
- * Traversal is depth-first, and assumes that target files will only be found at a single depth.
+ * Base class for {@link TableLocationKeyFinder location finders} that perform file system traversal to infer
+ * partitions.
  *
- * @implNote Column names will be legalized via {@link NameValidator#legalizeColumnName(String, Set)}.
+ * @param <TLK> The type of {@link TableLocationKey} to be generated
+ * @param <TARGET_FILE_TYPE> The type of files used to generate location keys, like a {@link URI} or a {@link Path}
  */
-public class KeyValuePartitionLayout<TLK extends TableLocationKey> implements TableLocationKeyFinder<TLK> {
+public abstract class KeyValuePartitionLayout<TLK extends TableLocationKey, TARGET_FILE_TYPE>
```

Comment:
The motivation for this refactoring was move all the file related logic from KeyValuePartitionLayout to FileKeyValuePartitionLayout and then URIStreamKeyValuePartitionLayout can extend from KeyValuePartitionLayout. 

I checked with Andy that KeyValuePartitionLayout is being used on enterprise side, so after this PR, they would need to do a rename from KeyValuePartitionLayout -> FileKeyValuePartitionLayout, which he is okay with. Once Ryan is okay with the new design, I will ping Andy and he said he can verify if their code works with this PR.